glean-core/src/debug.rs

```diff
@@ -33,63 +39,77 @@ pub struct DebugOptions {
     pub log_pings: DebugOption<bool>,
     /// Option to add the X-Debug-ID header to every ping request.
     pub debug_view_tag: DebugOption<String>,
+    /// Option to add the X-Source-Tags header to ping requests. This will allow the data
+    /// consumers to classify data depending on the applied tags.
+    pub sourge_tags: DebugOption<Vec<String>>,
 }
 
 impl DebugOptions {
     pub fn new() -> Self {
         Self {
-            log_pings: DebugOption::new(GLEAN_LOG_PINGS, None),
-            debug_view_tag: DebugOption::new(GLEAN_DEBUG_VIEW_TAG, Some(validate_debug_view_tag)),
+            log_pings: DebugOption::new(GLEAN_LOG_PINGS, get_bool_from_str, None),
+            debug_view_tag: DebugOption::new(GLEAN_DEBUG_VIEW_TAG, identity, Some(validate_tag)),
```

Comment:
Fun trick but this should work:

```suggestion
            debug_view_tag: DebugOption::new(GLEAN_DEBUG_VIEW_TAG, Some, Some(validate_tag)),
```

`Some` is not only the variant of this enum, but also a function that constructs that variant and thus can be used as such.

Reply:
WOHA. This is great :-)